conman/redirects/models.py

```diff
@@ -13,3 +15,9 @@ class RouteRedirect(Route):
     handler = handlers.RouteRedirectHandler.path()
     target = models.ForeignKey('routes.Route', related_name='+')
     permanent = models.BooleanField(default=False, blank=True)
+
+    def save(self, *args, **kwargs):
+        """Forbid setting target equal to self."""
+        if self.target_id == self.route_ptr_id:
+            raise ValidationError(_('A RouteRedirect cannot redirect to itself.'))
+        return super().save(*args, **kwargs)
```

Comment:
Or perhaps in the model's `clean` method? (I can never remember if that's invoked by forms.)